### ISSUE TITLE
CA1823 and IDE0052 overlap

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#4658]
The 2 analyzers above have overlapping behaviors, I would suggest to keep only one to reduce support.

**Comment by Evangelink** [comment]
Tagging @mavasani

**Comment by mavasani** [comment]
Agreed: https://github.com/dotnet/roslyn-analyzers/blob/b148fa89c974a72c28184931767835709f34d1fa/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidUnusedPrivateFields.cs#L31

As we discussed in the past, unnecessary code rules are a grey area on whether it belongs to style or quality, and we have decided to stick with them being IDE rules in unnecessary rules bucket as most existing ones are implemented there and have a better implementation: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/unnecessary-code-rules

I am fine with marking CA1823 deprecated in favor of IDE0052.

**Comment by mavasani** [comment]
Please target https://github.com/dotnet/roslyn-analyzers/tree/release/6.0.1xx-preview1 for any rule removal.

**Comment by Youssef1313** [comment]
Before removing CA1823, I'd suggest enabling it in roslyn repo to see if it detects any cases that IDE0052 doesn't
